extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -0,0 +1,168 @@
+						resolve(output.stdout);
+					}
+					catch (err) {
+						if (err instanceof Error) {
+							reject(err.message);
+						} else {
+							reject(err);
+						}
+					}
+				}
+			});
+		});
+	}
+
+	private getDeploymentProfileInfo(profileName: string): Thenable<DeploymentProfile> {
+		return new Promise<DeploymentProfile>((resolve, reject) => {
+			cp.exec(`azdata bdc config init --source ${profileName} --target ${profileName} --force`, { cwd: this.platformService.storagePath() }, (error, stdout, stderror) => {
+				if (error) {
+					reject(error.message);
+				} else {
+					const bdcJsonPromise = this.getJsonObjectFromFile(path.join(this.platformService.storagePath(), profileName, 'bdc.json'));
+					const controlJsonPromise = this.getJsonObjectFromFile(path.join(this.platformService.storagePath(), profileName, 'control.json'));
+					Promise.all([bdcJsonPromise, controlJsonPromise]).then((value) => {
+						const bdcJson = value[0];
+						const controlJson = value[1];
+						resolve({
+							name: profileName,
+							defaultDataSize: (<string>controlJson.spec.storage.data.size).replace('Gi', ''),
+							defaultLogSize: (<string>controlJson.spec.storage.logs.size).replace('Gi', ''),
+							defaultDataStorageClass: (<string>controlJson.spec.storage.data.className),
+							defaultLogsStorageClass: (<string>controlJson.spec.storage.logs.className),
+							master: <string>bdcJson.spec.resources[SqlServerMasterResourceName].spec.replicas,
+							data: <string>bdcJson.spec.resources[DataResourceName].spec.replicas,
+							compute: <string>bdcJson.spec.resources[ComputeResourceName].spec.replicas,
+							hdfs: <string>bdcJson.spec.resources[HdfsResourceName].spec.replicas,
+							nameNode: <string>bdcJson.spec.resources[NameNodeResourceName].spec.replicas,
+							spark: '0',
+							activeDirectory: false, // TODO: implement AD check
+							hadr: <string>bdcJson.spec.resources.master.spec.settings.sql[HadrEnabledSettingName],
+							includeSpark: <string>bdcJson.spec.resources[HdfsResourceName].spec.settings.spark.includeSpark,
+							gatewayPort: this.getEndpointPort(bdcJson.spec.resources.gateway.spec.endpoints, 'Knox'),
+							appProxyPort: this.getEndpointPort(bdcJson.spec.resources.appproxy.spec.endpoints, 'AppServiceProxy'),
+							sqlServerPort: this.getEndpointPort(bdcJson.spec.resources.master.spec.endpoints, 'Master'),
+							readableSecondaryPort: this.getEndpointPort(bdcJson.spec.resources.master.spec.endpoints, 'MasterSecondary', '31436'),
+							controllerPort: this.getEndpointPort(controlJson.spec.endpoints, 'Controller'),
+							bdcJson: bdcJson,
+							controlJson: controlJson
+						});
+					}).catch((error: Error) => {
+						reject(error.message);
+					});
+				}
+			});
+		});
+	}
+
+	private ensureWorkingDirectoryExists(): Thenable<void> {
+		return new Promise<void>((resolve, reject) => {
+			fs.access(this.platformService.storagePath(), (error) => {
+				if (error && error.code === 'ENOENT') {
```

Comment:
I'd suggest just using the fs.promises.access function and making this async so you can await on it